docs/concepts/deployments.md

```diff
@@ -16,621 +16,176 @@ search:
+Note that Prefect assumes that deployments reference flows defined _within Python files_. 
+This implies that flows defined within interactive REPLs or notebooks are not currently candidates for Prefect deployments. They are still valid flows that will be monitored by the API anytime they are run, but you must bear the responsibility for triggering them.
 
-### The Pull Action
+!!! info "Deployments do not contain code definitions"
+    Deployment metadata references code that exists in potentially diverse locations within your environment; this separation of concerns means that your flow code stays within your storage and execution infrastructure and never lives on the Prefect server or database.
 
-The pull section is the most important section within the `prefect.yaml` file as it contains instructions for preparing your flows for a deployment run.  These instructions will be executed each time a deployment created within this folder is run via a worker.
+    This is the heart of the Prefect hybrid model: there's a boundary between your proprietary assets and the Prefect backend (including [Prefect Cloud](/cloud/)). 
 
-There are three main types of steps that typically show up in a `pull` section:
+### Scheduling and parametrization
 
-- `set_working_directory`: this step simply sets the working directory for the process prior to importing your flow
-- `git_clone`: this step clones the provided repository on the provided branch
-- `pull_from_{cloud}`: this step pulls the working directory from a Cloud storage location (e.g., S3)
+One of the primary motivations for creating deloyments of flows is to remotely _schedule_ and _trigger_ them. 
+And just as flows can be called as functions with different input values, so can deployments be triggered or scheduled with different values through the use of parameters.
 
-!!! tip "Use block and variable references"
-    All [block and variable references](#templating-options) within your pull step will remain unresolved until runtime and will be pulled each time your deployment is run. This allows you to avoid storing sensitive information insecurely; it also allows you to manage certain types of configuration from the API and UI without having to rebuild your deployment every time.
+The five fields here capture the necessary metadata to perform such actions:
 
-Below is an example of how to use an existing `GitHubCredentials` block to clone a private GitHub repository:
+- **`schedule`**: a [schedule object](/concepts/schedules/); most of the convenient interfaces for creating deployments allow users to avoid creating this object themselves. For example, when [updating a deployment schedule in the UI](/concepts/schedules/#creating-schedules-through-the-ui) basic information such as a cron string or interval is all that's required.
```

Comment:
```suggestion
- **`schedule`**: a [schedule object](/concepts/schedules/).
Most of the convenient interfaces for creating deployments allow users to avoid creating this object themselves. 
For example, when [updating a deployment schedule in the UI](/concepts/schedules/#creating-schedules-through-the-ui) basic information such as a cron string or interval is all that's required.
```